Programs/_testembed.c

```diff
@@ -202,10 +202,9 @@ static int test_pre_initialization_sys_options(void)
     _Py_EMBED_PREINIT_CHECK("Initializing interpreter\n");
     _testembed_Py_Initialize();
     _Py_EMBED_PREINIT_CHECK("Check sys module contents\n");
-    PyRun_SimpleString("import sys; "
+    PyRun_SimpleString("import sys, warnings; "
                        "print('sys.warnoptions:', sys.warnoptions); "
                        "print('sys._xoptions:', sys._xoptions); "
-                       "warnings = sys.modules['warnings']; "
```

Comment:
Oh, this change is a bug related to _PySys_ReadPreInitOptions().

Reply:
_PySys_EndInit() calls _PySys_ReadPreInitOptions() to read options added by PySys_AddWarnOption() and PySys_AddXOption(). Problem: with this change, _PySys_ReadPreInitOptions() options are no longer added to interp->config.warnoptions, whereas _Py_InitializeMainInterpreter() checks interp->config.warnoptions is a non-empty list to decide if the "warnings" modules must be initialized.

Reply:
Ah, fixed by my second commit.